### PR TITLE
CADC-10865 - remove the global variables, refactor to MetaVisit approach

### PR DESCRIPTION
**Pull request by SharonGoliath**

### CODE REVIEW

File: gem2caom2/builder.py [hunk 1]
```diff
@@ -87,10 +85,13 @@ class GemObsIDBuilder(nbc.StorageNameBuilder):
...
     """
 
-    def __init__(self, config):
+    def __init__(self, config, metadata_reader, metadata):
         super(GemObsIDBuilder, self).__init__()
         self._config = config
-        self._logger = logging.getLogger(__name__)
+        self._metadata_reader = metadata_reader
+        self._metadata = metadata
+        self._metadata.reader = self._metadata_reader
```
Comment (at88mph) [inline gem2caom2/builder.py:93]: This (or the `_metadata_reader`) looks redundant.  Are both necessary?

File: gem2caom2/cleanup_augmentation.py [hunk 2]
```diff
@@ -128,7 +128,8 @@ def visit(observation, **kwargs):
         f'Completed cleanup for {observation.observation_id}. Removed '
         f'{artifact_count} artifacts and {plane_count} planes.'
     )
-    return {
+    result = {
```
Comment (at88mph) [inline gem2caom2/cleanup_augmentation.py:131]: Does this get used in a global way?  I can't see where it goes.

File: gem2caom2/preview_augmentation.py [hunk 3]
```diff
@@ -118,7 +118,8 @@ def visit(observation, **kwargs):
     logging.info(
         f'Completed preview augmentation for {observation.observation_id}.'
     )
-    return {'artifacts': count}
+    result = {'artifacts': count}
```
Comment (at88mph) [inline gem2caom2/preview_augmentation.py:121]: Now I assume this is getting used elsewhere. :)

**Review by at88mph — CHANGES_REQUESTED** [review]

File: gem2caom2/builder.py (hunk 1 above)
Comment (SharonGoliath) [reply gem2caom2/builder.py:93]: The self._metadata class is for looking up values in JSON records retrieved from archive.gemini.edu.

The self._metadata_reader class is for retrieving the JSON records from archive.gemini.edu.

So, yes, both are necessary. Would you prefer different names?

Comment (at88mph) [reply gem2caom2/builder.py:93]: No.  If that object contained state I could see some code smell there.  Is it just a stateless reader?

File: gem2caom2/preview_augmentation.py (hunk 3 above)
Comment (SharonGoliath) [reply gem2caom2/preview_augmentation.py:121]: Not right now - see comment above.

**Review by at88mph — APPROVED** [review]

File: gem2caom2/builder.py (hunk 1 above)
Comment (SharonGoliath) [reply gem2caom2/builder.py:93]: self._metadata is a stateless reader. Was that the data member you were referring to?

**Pull request merged**